### PR TITLE
DOC: use intersphinx mapping in pandas-compat ext

### PR DESCRIPTION
**Pull request by raybellwaves**
~~If https://github.com/rapidsai/cudf/pull/15704 is merged~~

This PR changes the header in the admonition (pandas compat box) to be hyperlinked to the pandas docs instead of just text. See https://raybellwaves.github.io/compatsphinxext/compat.html which is the docs of a minimal repo where I have been testing

### CODE REVIEW

**Review by bdice — COMMENTED** [review]
I have one suggestion.

File: python/cudf/cudf/core/groupby/groupby.py [hunk 1]
```diff
@@ -736,7 +736,8 @@ def _reduce(
             Computed {op} of values within each group.
 
         .. pandas-compat::
-            **{cls}.{op}**
+            :meth:`pandas.core.groupby.DataFrameGroupBy.{op}`,
+             :meth:`pandas.core.groupby.SeriesGroupBy.{op}
```
Comment (charlesbluca) [inline python/cudf/cudf/core/groupby/groupby.py:740]: Assuming these missed backticks are a typo?

```suggestion
             :meth:`pandas.core.groupby.SeriesGroupBy.{op}`
```

Also applies to the updates to `GroupBy.apply` and `shift` below

Comment (raybellwaves) [reply python/cudf/cudf/core/groupby/groupby.py:740]: I think so. I'll take a look. Thanks

**Comment by raybellwaves** [comment]
Apologies for lack of response here. Using this comments to give an update on where I am with this. The goal here is to switch https://docs.rapids.ai/api/cudf/stable/user_guide/pandascompat/ to the use the new sphinx extension which links back to pandas docs.

I've listed the changes that have to be made (we may as well try and get them in one sweep) by listing the order they appear in https://docs.rapids.ai/api/cudf/stable/user_guide/pandascompat/. A check means the changes render as expected. An empty box means a note is missing.

Looking at the list below there are quite a few dups. I'll look to see if I can remove those before continuing here.

 - [ ] DataFrame.transpose, DataFrame.T
 - [ ] DataFrame.agg
 - [ ] DataFrame.all, Series.all
 - [ ] DataFrame.any, Series.any
 - [ ] DataFrame.count
 - [ ] DataFrame.diff
 - [ ] DataFrame.empty, Series.empty
 - [ ] DataFrame.eval
 - [ ] DataFrame.from_arrow
 - [ ] DataFrame.interleave_columns
 - [ ] DataFrame.join
 - [ ] DataFrame.kurtosis
 - [ ] DataFrame.kurtosis (not sure why this appears twice)
 - [ ] DataFrame.max, Series.max
 - [ ] DataFrame.median, Series.median
 - [ ] DataFrame.merge
 - [ ] DataFrame.min, Series.min
 - [ ] DataFrame.mode
 - [ ] DataFrame.nlargest
 - [ ] DataFrame.nsmallest
 - [ ] DataFrame.product, Series.product
 - [ ] DataFrame.product, Series.product (not sure why this appears twice)
 - [ ] DataFrame.quantile
 - [ ] DataFrame.query
 - [ ] DataFrame.reindex
 - [ ] DataFrame.rename
 - [ ] DataFrame.replace, Series.replace
 - [ ] DataFrame.resample, Series.resample
 - [ ] DataFrame.sample, Series.sample
 - [ ] DataFrame.skew, Series.skew, Frame.skew
 - [ ] DataFrame.sort_index, Series.sort_index
 - [ ] DataFrame.sort_values, Series.sort_values
 - [ ] DataFrame.std, Series.std
 - [ ] DataFrame.sum, Series.sum
 - [ ] DataFrame.transpose, DataFrame.T
 - [ ] DataFrame.truncate, Series.truncate
 - [ ] DataFrame.var, Series.var
 - [ ] DataFrame.where, Series.where
 - [ ] DataFrame.all, Series.all
 - [ ] DataFrame.any, Series.any (not sure why this appears twice)
 - [ ] DataFrame.max, Series.max (not sure why this appears twice)
 - [ ] DataFrame.min, Series.min (not sure why this appears twice)
 - [ ] DataFrame.where, Series.where
 - [ ] DataFrame.all, Series.all (not sure why this appears twice)
 - [ ] DataFrame.any, Series.any (not sure why this appears twice)
 - [ ] Series.count
 - [ ] Series.cov
 - [ ] DataFrame.empty, Series.empty
 - [ ] DataFrame.kurtosis (not sure they appears three times)
 - [ ] DataFrame.kurtosis (not sure they appears four times)
 - [ ] Series.map
 - [ ] DataFrame.max, Series.max (not sure why this appears three times)
 - [ ] DataFrame.median, Series.median (not sure why this appears twice)
 - [ ] DataFrame.min, Series.min (not sure why this appears three times)
 - [ ] DataFrame.product, Series.product (not sure why this appears three times)
 - [ ] DataFrame.product, Series.product (not sure why this appears four times)
 - [ ] Series.reindex
 - [ ] Series.rename
 - [ ] DataFrame.replace, Series.replace (not sure why this appears twice)
 - [ ] DataFrame.resample, Series.resample (not sure why this appears twice)
 - [ ] DataFrame.sample, Series.sample (not sure why this appears twice)
 - [ ] DataFrame.skew, Series.skew, Frame.skew
 - [ ] DataFrame.sort_index, Series.sort_index (not sure why this appears twice)
 - [ ] Series.sort_values
 - [ ] DataFrame.std, Series.std (not sure why this appears twice)
 - [ ] DataFrame.sum, Series.sum (not sure why this appears twice)
 - [ ] DataFrame.truncate, Series.truncate
 - [ ] DataFrame.var, Series.var
 - [ ] DataFrame.where, Series.where (not sure why this appears three times)
 - [ ] ListMethods.sort_values
 - [ ] StringMethods.contains
 - [ ] StringMethods.count
 - [ ] StringMethods.endswith
 - [ ] StringMethods.extract
 - [ ] StringMethods.findall
 - [ ] StringMethods.match
 - [ ] StringMethods.partition
 - [ ] StringMethods.replace
 - [ ] DataFrameGroupBy.idxmax
 - [ ] DataFrameGroupBy.idxmin
 - [ ] DataFrameGroupBy.nunique
 - [ ] GroupBy.shift
 - [ ] GroupBy.apply
 - [ ] GroupBy.first
 - [ ] GroupBy.idxmax
 - [ ] GroupBy.idxmin
 - [ ] GroupBy.last
 - [ ] GroupBy.max
 - [ ] GroupBy.mean
 - [ ] GroupBy.median
 - [ ] GroupBy.min
 - [ ] GroupBy.nunique
 - [ ] GroupBy.prod
 - [ ] GroupBy.shift
 - [ ] GroupBy.sum
 - [ ] GroupBy.nunique
 - [ ] series.DatetimeProperties.strftime
 - [ ] DataFrame.merge
 - [ ] cudf.to_numeric

**Comment by bdice** [comment]
@raybellwaves I merged in the upstream a few days ago to check on the CI output. There were a few failures in the previous doc build ([logs link](https://github.com/rapidsai/cudf/actions/runs/9829007343/job/27134056169#step:9:4399)):

```
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.T:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty

/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.from_arrow:40: WARNING: py:meth reference target not found: pandas.DataFrame.from_arrow
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.interleave_columns:17: WARNING: py:meth reference target not found: pandas.DataFrame.interleave_columns
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.transpose:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/column/lists.py:docstring of cudf.core.column.lists.ListMethods.sort_values:41: WARNING: py:meth reference target not found: pandas.Series.list.sort_values
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/tools/numeric.py:docstring of cudf.core.tools.numeric.to_numeric:78: WARNING: py:meth reference target not found: pandas.to_numeric
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.T:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.from_arrow:40: WARNING: py:meth reference target not found: pandas.DataFrame.from_arrow
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.interleave_columns:17: WARNING: py:meth reference target not found: pandas.DataFrame.interleave_columns
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.transpose:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/column/lists.py:docstring of cudf.core.column.lists.ListMethods.sort_values:41: WARNING: py:meth reference target not found: pandas.Series.list.sort_values
/opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/tools/numeric.py:docstring of cudf.core.tools.numeric.to_numeric:78: WARNING: py:meth reference target not found: pandas.to_numeric
```

I see you broke off #16258, but I would be happy to merge this PR with a larger set of changes if you can fix those docs errors. If you need help fixing the docs errors, please ask!

**Comment by raybellwaves** [comment]
> @raybellwaves I merged in the upstream a few days ago to check on the CI output. There were a few failures in the previous doc build ([logs link](https://github.com/rapidsai/cudf/actions/runs/9829007343/job/27134056169#step:9:4399)):
> 
> ```
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.T:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
> 
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.from_arrow:40: WARNING: py:meth reference target not found: pandas.DataFrame.from_arrow
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.interleave_columns:17: WARNING: py:meth reference target not found: pandas.DataFrame.interleave_columns
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.transpose:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/column/lists.py:docstring of cudf.core.column.lists.ListMethods.sort_values:41: WARNING: py:meth reference target not found: pandas.Series.list.sort_values
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/tools/numeric.py:docstring of cudf.core.tools.numeric.to_numeric:78: WARNING: py:meth reference target not found: pandas.to_numeric
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.T:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.DataFrame.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.from_arrow:40: WARNING: py:meth reference target not found: pandas.DataFrame.from_arrow
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.interleave_columns:17: WARNING: py:meth reference target not found: pandas.DataFrame.interleave_columns
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/dataframe.py:docstring of cudf.core.dataframe.DataFrame.transpose:12: WARNING: py:meth reference target not found: pandas.DataFrame.T
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.DataFrame.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/series.py:docstring of cudf.Series.empty:64: WARNING: py:meth reference target not found: pandas.Series.empty
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/column/lists.py:docstring of cudf.core.column.lists.ListMethods.sort_values:41: WARNING: py:meth reference target not found: pandas.Series.list.sort_values
> /opt/conda/envs/docs/lib/python3.11/site-packages/cudf/core/tools/numeric.py:docstring of cudf.core.tools.numeric.to_numeric:78: WARNING: py:meth reference target not found: pandas.to_numeric
> ```
> 
> I see you broke off #16258, but I would be happy to merge this PR with a larger set of changes if you can fix those docs errors. If you need help fixing the docs errors, please ask!

Sounds good.

File: python/cudf/cudf/core/column/lists.py [hunk 2]
```diff
@@ -643,9 +643,10 @@ def sort_values(
         dtype: list
 
         .. pandas-compat::
-            **ListMethods.sort_values**
+            :meth:`pandas.Series.list.sort_values`
 
-            The ``inplace`` and ``kind`` arguments are currently not supported.
+            This method does not exist in pandas but it can be run
+            as ``...``.
```
[the remark this answers is not in the document]
Comment (raybellwaves) [reply python/cudf/cudf/core/column/lists.py]: should be fixed now. Removed the link as me so the warning shouldn't pop up in the docs build

File: python/cudf/cudf/core/groupby/groupby.py (hunk 1 above)
Comment (raybellwaves) [reply python/cudf/cudf/core/groupby/groupby.py]: should be fixed now

**Comment by raybellwaves** [comment]
Should be ready to test again

**Comment by raybellwaves** [comment]
i'm still missing`cudf.core.window.ExponentialMovingWindow`

**Comment by raybellwaves** [comment]
should be ready to test again. Apologies for the multiple commits

File: python/cudf/cudf/core/dataframe.py [hunk 3]
```diff
@@ -5449,10 +5449,11 @@ def from_arrow(cls, table):
         2  3  6
 
         .. pandas-compat::
-            **DataFrame.from_arrow**
+            `pandas.DataFrame.from_arrow`
 
-            -   Does not support automatically setting index column(s) similar
-                to how ``to_pandas`` works for PyArrow Tables.
+            This method does not exit in pandas but it is simlar to
```
Comment (vyasr) [inline python/cudf/cudf/core/dataframe.py:5454]:
```suggestion
            This method does not exist in pandas but it is similar to
```

**Review by vyasr — APPROVED** [review]
Looks great, thanks!

**Pull request merged**